src/config.ts

```diff
@@ -1,22 +1,169 @@
-const VALID_ENVIRONMENTS = ['local', 'staging', 'prod'];
-if (!VALID_ENVIRONMENTS.includes(process.env.ENV)) {
-  throw new Error(`Invalid settings: ENV=${process.env.ENV}`);
-}
+const VALID_ENVIRONMENTS: Set<string> = new Set(['local', 'staging', 'prod']);
 
-if (
-  process.env.ENV === 'staging' &&
-  process.env.S3_BUCKET_NAME !== 'proavalon-staging'
-) {
-  throw new Error(
-    `Invalid settings: ENV=staging S3_BUCKET_NAME=${process.env.S3_BUCKET_NAME}`,
-  );
-}
+class Config {
+  private readonly env: string = process.env.ENV;
```

Comment:
Can we have a typescript `type` for env pls? 
E.g. `type Env = 'local' | 'staging' | 'prod'`